### DISCUSSION TITLE
Is it possible to use for different accounts in one Laravel Application?

### DISCUSSION THREAD

**Discussion by yergunes** [gtmassey/Analytics#50] [Q&A] [answered]
I would like to connect multiple accounts on a single Laravel application, is there a way to send credentials and property ID as variables?

**Comment by Plytas** [comment, marked as answer]
Hey @yergunes.

It actually is possible to use multiple service accounts albeit not as straight forward. You can change the config value before starting a new query and it should resolve to a new account:

```PHP
config(['analytics.credentials.file' => 'path/to/credentials1.json']);
config(['property_id' => 'property1']);

Analytics::query()
...
->run();

config(['analytics.credentials.file' => 'path/to/credentials2.json']);
config(['property_id' => 'property1']);

Analytics::query()
...
->run();
```

You can use any of the available authentication ways outlined in [README](https://github.com/gtmassey/laravel-analytics#env).

Let me know if that helps in your case. We can try look for better ways to achieve it as well.

**Reply by yergunes** [reply]
Thank you for quick reply, this will be very helpful.
I think this lines should be

`config(['analytics.property_id' => 'property1']);`

**Reply by Plytas** [reply]
You are correct. I typed that from the top of my head without verifying it 😄
